apps/prairielearn/src/lib/editors.ts

```diff
@@ -192,20 +192,37 @@ export function getUniqueNames({
  * `defaultValue` may be either a value to compare directly with `===`, or a function
  * that accepts a value and returns a boolean to indicate if it should be considered
  * a default value.
+ *
+ * You should set `isUIBoolean` to true if the property is a UI boolean that controls the visibility of another field.
+ * For example, `beforeDateEnabled` is a UI boolean that controls the visibility of the `beforeDate` field.
  */
-export function propertyValueWithDefault(existingValue: any, newValue: any, defaultValue: any) {
+export function propertyValueWithDefault(
+  existingValue: any,
+  newValue: any,
+  defaultValue: any,
+  { isUIBoolean = false }: { isUIBoolean?: boolean } = {},
+) {
   const isExistingDefault =
     typeof defaultValue === 'function'
       ? defaultValue(existingValue)
       : existingValue === defaultValue;
   const isNewDefault =
     typeof defaultValue === 'function' ? defaultValue(newValue) : newValue === defaultValue;
 
+  // If this is a UI boolean where the default value is false, we want to write that out as false, not as undefined.
+  const writeFalse = isUIBoolean && defaultValue === false && newValue === false;
+  if (writeFalse) {
+    return false;
+  }
+
+  // If the existing value is undefined, we want to write the new value if it differs from the default value.
   if (existingValue === undefined) {
     if (!isNewDefault) {
       return newValue;
     }
+    // Otherwise, we want to write undefined.
   } else {
+    // If the existing value is not the default value, and the new value is the default value, we want to write undefined.
```

Comment:
Nit: I don't understand the point of this comment (and tbh the above ones), this seems to repeat verbatim what the following line does without any extra context/explanation?

Reply:
Fixed in e9a0f59b3e9da87dbfdfb8caaf891ffa5e94ddbe.